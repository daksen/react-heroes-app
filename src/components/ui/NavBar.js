import React, { useContext } from "react";
import { Link, NavLink, useHistory } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

export const Navbar = () => {
  const { user, dispatch } = useContext(AuthContext);
  const history = useHistory();

  const handleLogout = () => {
    dispatch({
      type: types.logout,
    });
    history.replace('/login');
  }

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <Link className="navbar-brand h1 mb-0" to="/">
          HEROES-APP
        </Link>
        <div className="navbar-collapse">
          <div className="navbar-nav me-auto">
            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/marvel"
            >
              Marvel
            </NavLink>
            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/dc"
            >
              DC
            </NavLink>
            <NavLink
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to="/search"
            >
              Search
            </NavLink>
          </div>
          <div className="navbar-nav">
            <span className="nav-item nav-link text-info m-0">
              { user.username }
            </span>
            <button
              className="nav-item nav-link btn"
              onClick={ handleLogout }
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  )
}
