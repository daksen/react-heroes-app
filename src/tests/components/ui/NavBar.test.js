import React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import { MemoryRouter, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/NavBar';
import { types } from '../../../types/types';


describe('Pruebas en <NavBar />', () => {
  // CUSTOM HISTORY
  const historyMock = {
    location: {},
    push: jest.fn(),
    replace: jest.fn(),
    createHref: jest.fn(),
    listen: jest.fn()
  };

  const contextValues = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      username: "Test"
    }
  }

  const wrapper = mount(
    <AuthContext.Provider value={ contextValues }>
      <MemoryRouter>
        <Router history={ historyMock }>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  afterEach(() => {
    jest.clearAllMocks();
  });
  
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("span.text-info").text().trim()).toBe("Test")
  });

  test('debe de llamar el logout y usar el history', () => {
    wrapper.find('button').prop('onClick')();
    expect(contextValues.dispatch).toHaveBeenCalledWith({
      type: types.logout
    });
    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });

});
