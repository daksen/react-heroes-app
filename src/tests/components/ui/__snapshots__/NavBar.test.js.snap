// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar /> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand h1 mb-0"
      onClick={[Function]}
    >
      HEROES-APP
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav me-auto"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
      <div
        className="navbar-nav"
      >
        <span
          className="nav-item nav-link text-info m-0"
        >
          Test
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</nav>
`;
